### PR TITLE
CLI: Add release-branch to jetpack release

### PR DESCRIPTION
**Pull request by sdixon194**
#### Changes proposed in this Pull Request:

This PR adds `release-branch` to the `jetpack release` command so you can create a release branch using the CLI tool.

#### Jetpack product discussion

#21160

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

Kinda tricky to test, since the script has a lot of error handling, but ideally:

* Run `jetpack release` and follow the prompts to create a release branch for a project.
* It should automatically suggest a potential version, but you can enter your own.
* Try passthroughs - `jetpack release plugins/jetpack -b release-branch` should run the release branch script for 10.4-beta, etc.
* Don't push any branches you create.

Might be easier to merge and put to the test during the 10.4 release later this month. Although I notice the 10.4 branch was already created.

### CODE REVIEW

File: tools/cli/commands/release.js [hunk 1]
```diff
@@ -136,10 +141,20 @@ export async function scriptRouter( argv ) {
 				      jetpack release ${ argv.project } release-branch \n`.replace( /^\t+/gm, '' );
 			break;
 		case 'release-branch':
+			argv = await getReleaseVersion( argv );
+			argv.script = `tools/create-release-branch.sh`;
+			argv.scriptArgs = [ argv.project, argv.version ];
+			argv.next = `Finished! Next: 
+				  - Once the branch is pushed, GitHub Actions will build and create a branch jetpack/branch-x.x over at: https://github.com/Automattic/jetpack-production
+				  - jetpack-production/branch-x.x is the built version for release, and will be the branch that is tagged in GitHub and pushed to svn in WordPress.org.
+				  - When changes are pushed to jetpack/branch-x.x, GitHub Actions takes care of building/mirroring to the jetpack-production repo.
```
Comment (anomiex) [inline tools/cli/commands/release.js:150]: All the references to "jetpack/branch-x.x" and jetpack-production are specific to Jetpack-the-plugin.

The bit about "jetpack-production/branch-x.x" is confusing to me, there's nothing by that name anywhere as far as I know. It seems like you're referring to the "jetpack/branch-x.x" branch in the jetpack-production mirror repo?

File: tools/cli/commands/release.js [hunk 2]
```diff
@@ -75,15 +75,20 @@ export async function releaseCli( argv ) {
...
+		( argv.script === 'changelog' || argv.script === 'release-branch' )
+	) {
 		argv = await promptBeta( argv );
 	}
 
 	// Get the info we need for the script.
 	await scriptRouter( argv );
 
 	// Run the script.
+
```
[the remark this answers is not in the document]
Comment (sdixon194) [reply tools/cli/commands/release.js:91]: vestigial space!

File: tools/cli/commands/release.js (hunk 1 above)
Comment (sdixon194) [reply tools/cli/commands/release.js:149]: > All the references to "jetpack/branch-x.x" and jetpack-production are specific to Jetpack-the-plugin.

Ah you're right, nice catch here. I'll update this to be inclusive of all plugins. This copy was from the Jetpack Releases Field Guide page.

>It seems like you're referring to the "jetpack/branch-x.x" branch in the jetpack-production mirror repo?

Yeah this is exactly it. It's basically just "keep an eye on the mirror repo."

File: tools/cli/commands/release.js [hunk 3]
```diff
@@ -144,10 +144,10 @@ export async function scriptRouter( argv ) {
 			argv.script = `tools/create-release-branch.sh`;
 			argv.scriptArgs = [ argv.project, argv.version ];
 			argv.next = `Finished! Next: 
-				  - Once the branch is pushed, GitHub Actions will build and create a branch jetpack/branch-x.x over at: https://github.com/Automattic/jetpack-production
-				  - jetpack-production/branch-x.x is the built version for release, and will be the branch that is tagged in GitHub and pushed to svn in WordPress.org.
-				  - When changes are pushed to jetpack/branch-x.x, GitHub Actions takes care of building/mirroring to the jetpack-production repo.
-				  - You will likely want to start a new release cycle like so:
+				  - Once the branch is pushed, GitHub Actions will build and create a branch on your plugin's mirror repo.
+				  - That mirror repo branch will be the branch that is tagged in GitHub and pushed to svn in WordPress.org.
+				  - When changes are pushed to the release branch that was just created, GitHub Actions takes care of building/mirroring to the jetpack-production repo.
```
Comment (anomiex) [inline tools/cli/commands/release.js:149]: Missed one
```suggestion
				  - When changes are pushed to the release branch that was just created, GitHub Actions takes care of building/mirroring to the mirror repo.
```

**Review by anomiex — COMMENTED** [review]
Works in manual testing. So other than the inline comment and the need for a master merge, looks good to go.

**Comment by sdixon194** [comment]
Thanks for taking another look! Should be good for approval now 👍

**Review by anomiex — APPROVED** [review]

**Pull request merged**